st2reactor/st2reactor/container/utils.py

```diff
@@ -28,14 +28,22 @@
 
 def create_trigger_instance(trigger, payload, occurrence_time):
     """
+    This creates a trigger instance object given trigger and payload.
+    Trigger can be just a string reference (pack.name) or a ``dict``
+    containing  'type' and 'parameters'.
+
     :param trigger: Dictionary with trigger query filters.
     :type trigger: ``dict``
 
     :param payload: Trigger payload.
-    :type payload: ``dict``
+    :type payload: ``dict``p
     """
     # TODO: This is nasty, this should take a unique reference and not a dict
-    trigger_db = TriggerService.get_trigger_db(trigger)
+    if isinstance(trigger, str):
```

Comment:
Probably safer to use `six.string_types`, otherwise we will have a bad time if trigger is unicode :P